planetiler-core/src/main/java/com/onthegomap/planetiler/ForwardingProfile.java

```diff
@@ -81,19 +81,22 @@ protected ForwardingProfile(Handler... handlers) {
   }
 
   private boolean caresAboutLayer(String layer) {
+    var dependencies = dependsOnLayer();
     return ((onlyLayers.isEmpty() || onlyLayers.contains(layer)) && !excludeLayers.contains(layer)) ||
       this.handlers.stream()
-        .filter(h -> h instanceof HandlerForLayer)
-        .map(h -> (HandlerForLayer) h)
-        .anyMatch(existing -> dependsOnLayer(existing.name(), layer));
+        .filter(HandlerForLayer.class::isInstance)
+        .map(HandlerForLayer.class::cast)
+        .filter(l -> !l.name().equals(layer))
+        .anyMatch(
+          existing -> dependencies.containsKey(existing.name()) && dependencies.get(existing.name()).contains(layer));
   }
 
   public boolean caresAboutLayer(Object obj) {
     return !(obj instanceof HandlerForLayer l) || caresAboutLayer(l.name());
   }
 
-  public boolean dependsOnLayer(String dependent, String dependency) {
-    return false;
+  public Map<String, List<String>> dependsOnLayer() {
```

Comment:
Minor: add javadoc to make it clear this should return a map from layer to list of layer names that it depends on (not the other way around)

Reply:
javadoc added